#Vikas Bansal 25 Nov 2019
#Foundin project. Raw expression matrix were generated by CellRanger pipeline using GTF from Kendall. 

set.seed(786)
setwd("/data/vikas/FOUNDIN_scRNA/")

library(Seurat)
library(devtools)
library(ggplot2)
library(cowplot)

library(dplyr)
library(readxl)

library(stringi)
library(purrr)
library(reshape)


pathto.outPlots <- "/data/vikas/FOUNDIN_scRNA/OutputPlots/"
pathto.outData <- "/data/vikas/FOUNDIN_scRNA/OutputData/"
outName <- "iPSCsDopa27"

dirs <- list.dirs(path="/data/vikas/FOUNDIN_scRNA/Counts_copy/")
Gene_results_file_names <- dirs[ grepl("filtered_feature_bc_matrix", dirs) ]


#Read all the count files generated by Cell Ranger pipeline (in filtered_feature_bc_matrix folder)
test_data <- Read10X(data.dir = Gene_results_file_names[1])


Sample_name <- unlist(strsplit(Gene_results_file_names[1],"/"))[[7]]

ALL_WITH_COUNT <- as.data.frame(test_data)

colnames(ALL_WITH_COUNT) <- paste(colnames(ALL_WITH_COUNT),Sample_name, sep="_")

All_joined_countMatrixTemp <- (ALL_WITH_COUNT)

for (i in 2:length(Gene_results_file_names)){
  cat("loop", i, "\n")
  
  
  test_data <- Read10X(data.dir = Gene_results_file_names[i])
  
  
  Sample_name <- unlist(strsplit(Gene_results_file_names[i],"/"))[[7]]
  
  ALL_WITH_COUNT <- as.data.frame(test_data)
  
  colnames(ALL_WITH_COUNT) <- paste(colnames(ALL_WITH_COUNT),Sample_name, sep="_")
  
  
  All_joined_countMatrix <- cbind(All_joined_countMatrixTemp,ALL_WITH_COUNT)
  
  All_joined_countMatrixTemp <- All_joined_countMatrix
}



#Read in the meta DATA info for all the samples/cell lines
metaDataOurs <-  data.frame(matrix(nrow=ncol(All_joined_countMatrix),ncol=2))

metaDataOurs[,1] <- colnames(All_joined_countMatrix) 

metaDataOurs[,2] <- (sapply(stri_split_fixed(str = metaDataOurs[,1], pattern = "_", n = 2),"[[",2))

colnames(metaDataOurs)[1] <- "WholeID"
colnames(metaDataOurs)[2] <- "SampleID"


Petertable <- read_excel("MetaInfo/PPMI_IPS_overview_MAPT_peter.xlsx", sheet = 1)

Petertablev2 <- (as.data.frame(Petertable))
Petertablev3 <- Petertablev2[grep("CDI",Petertablev2[,"Barcode_DZNE"]),]


Petertablev3$Barcode_last4 <- (stri_sub(Petertablev3$Barcode_DZNE,-4,-1))

Petertablev3$SampleID <- (paste0("SCRN_PPMI",Petertablev3$PPMI_ID, "_", Petertablev3$Barcode_last4,"_da65"))


metaInfoAll <- (merge(metaDataOurs, Petertablev3, by="SampleID"))

rownames(metaInfoAll) <- metaInfoAll$WholeID

rm(list = c('test_data','ALL_WITH_COUNT','All_joined_countMatrixTemp','All_joined_countMatrix','metaDataOurs','Petertablev2','Petertablev3','Petertable'))


##Now we start with integeration


iPSCsDopa <- CreateSeuratObject(All_joined_countMatrix, meta.data = metaInfoAll, min.cells = 100, min.features = 1000)


iPSCsDopa[["percent.mt"]] <- PercentageFeatureSet(iPSCsDopa, pattern = "^MT-")


##Remove the cells with >=50% of reads/UMIs coming for mitochondrial genes
iPSCsDopa <- subset(iPSCsDopa, subset =  percent.mt < 50)



iPSCsDopa.list <- SplitObject(iPSCsDopa, split.by = "SampleID")

##Store the fileterd object for later use for not integration analysis
iPSCsDopa_WoInt <- iPSCsDopa


for (i in 1:length(iPSCsDopa.list)) {
  iPSCsDopa.list[[i]] <- SCTransform(iPSCsDopa.list[[i]])
}


iPSCsDopa.features <- SelectIntegrationFeatures(object.list = iPSCsDopa.list, nfeatures = 3000)
iPSCsDopa.list <- PrepSCTIntegration(object.list = iPSCsDopa.list, anchor.features = iPSCsDopa.features)





reference.list <- iPSCsDopa.list[unique(metaInfoAll[,"SampleID"])]

#Take top 3 datset with highest number of cells
reference_dataset <- match(names(tail(sort(table(iPSCsDopa$SampleID)),3)),names(reference.list))

iPSCsDopa.anchors <- FindIntegrationAnchors(object.list = iPSCsDopa.list, normalization.method = "SCT", 
                                            anchor.features = iPSCsDopa.features, reference = reference_dataset)
iPSCsDopa.integrated <- IntegrateData(anchorset = iPSCsDopa.anchors, normalization.method = "SCT")

rm(list = c('iPSCsDopa','iPSCsDopa.anchors','iPSCsDopa.list','reference.list'))

iPSCsDopa.integrated <- RunPCA(iPSCsDopa.integrated, npcs = 30)

iPSCsDopa.integrated <- RunTSNE(iPSCsDopa.integrated, nthreads = 40, max_iter = 2000, dims = 1:30)





iPSCsDopa.integrated <- RunUMAP(iPSCsDopa.integrated, dims = 1:30)




DefaultAssay(iPSCsDopa.integrated) <- "integrated"

iPSCsDopa.integrated <- FindNeighbors(iPSCsDopa.integrated, reduction = "pca", dims = 1:30, nn.eps = 0.5)
iPSCsDopa.integrated <- FindClusters(iPSCsDopa.integrated, resolution = 0.1, n.start = 10)



p2 <- DimPlot(iPSCsDopa.integrated, reduction = "umap", label = TRUE)


pdf(paste(pathto.outPlots,"UMAP_integrated.pdf"), width=10, height=10)
plot_grid(p2)
dev.off()




DefaultAssay(iPSCsDopa.integrated) <- "RNA"


##Normalise the data in RNA assay for plotting and differetial expression
iPSCsDopa.integrated <- NormalizeData(iPSCsDopa.integrated, verbose = FALSE)

DefaultAssay(iPSCsDopa.integrated) <- "integrated"

saveRDS(iPSCsDopa.integrated,file = paste0(pathto.outData,outName,"_iPSCsDopa.integrated.RDS"))




#*****************Generate a heatmap to see what percentage of cells are coming from each sample to a cluster or vice versa*****************
NumberOfPerSampleCellsInCluster <- as.list(table(paste0(iPSCsDopa.integrated$SampleID,";",iPSCsDopa.integrated$seurat_clusters)))

NumberOfPerSampleCellsInClusterDf <- (data.frame(matrix(nrow=length(unique(iPSCsDopa.integrated$SampleID)),ncol=length(unique(iPSCsDopa.integrated$seurat_clusters)))))

rownames(NumberOfPerSampleCellsInClusterDf) <- unique(iPSCsDopa.integrated$SampleID)

colnames(NumberOfPerSampleCellsInClusterDf) <- paste0("Cluster",sort(unique(iPSCsDopa.integrated$seurat_clusters)))


for (RowNum in 1:nrow(NumberOfPerSampleCellsInClusterDf)){
  
  
  for (ColNum in 1:ncol(NumberOfPerSampleCellsInClusterDf)){
    ValToPut <- NumberOfPerSampleCellsInCluster[match(gsub("Cluster","",paste0(rownames(NumberOfPerSampleCellsInClusterDf)[RowNum],";",
                                                                               colnames(NumberOfPerSampleCellsInClusterDf)[ColNum])),names(NumberOfPerSampleCellsInCluster))]
    if(is.numeric(unlist(ValToPut))){
      NumberOfPerSampleCellsInClusterDf[RowNum,ColNum] <-ValToPut
    } else {NumberOfPerSampleCellsInClusterDf[RowNum,ColNum] <- 0}
    
  }
  
}

NumberOfPerSampleCellsInClusterDfV2 <- NumberOfPerSampleCellsInClusterDf

NumberOfPerSampleCellsInClusterDf$TotalCells <- table(iPSCsDopa.integrated$SampleID)




NumberOfPerSampleCellsInClusterDfV2Percent <- (NumberOfPerSampleCellsInClusterDfV2/NumberOfPerSampleCellsInClusterDf$TotalCells)*100




NumberOfPerSampleCellsInClusterDfV2PercentV2 <- NumberOfPerSampleCellsInClusterDfV2Percent
colnames(NumberOfPerSampleCellsInClusterDfV2PercentV2) <- paste0(colnames(NumberOfPerSampleCellsInClusterDfV2Percent)," (",table(iPSCsDopa.integrated$seurat_clusters),")")


pdf(paste(pathto.outPlots,"Heatmap_PercentOfPerSampleCellsInCluster_integrated.pdf"), width=10, height=10)

#par(mar=c(17,14,14,12)) 
heatmap(as.matrix(t(NumberOfPerSampleCellsInClusterDfV2PercentV2)), Rowv=NA, Colv=NA, margins = c(15,10))

dev.off()




NumberOfPerSampleCellsInClusterDfV3 <- t(NumberOfPerSampleCellsInClusterDfV2)
NumberOfPerClusterCellsFromSamplesDfV2Percent <- (t((NumberOfPerSampleCellsInClusterDfV3/unlist(colSums(NumberOfPerSampleCellsInClusterDfV2)))*100))

pdf(paste(pathto.outPlots,"Heatmap_PercentOfPerClusterCellsFromSamples_integrated.pdf"), width=10, height=10)

#par(mar=c(17,14,14,12)) 
heatmap(as.matrix(t(NumberOfPerClusterCellsFromSamplesDfV2Percent)), Rowv=NA, Colv=NA, margins = c(15,10))

dev.off()

**********************************************************************************************************************************************





##Without integration

#Take iPSCsDopa from abpove so that filetring is same.
#iPSCsDopa_WoInt <- CreateSeuratObject(All_joined_countMatrix, project = "Foundin_woIntegration", min.cells = 3, min.features = 200, meta.data = metaInfoAll)




iPSCsDopa_WoInt <- NormalizeData(iPSCsDopa_WoInt, normalization.method = "LogNormalize", scale.factor = 10000)



iPSCsDopa_WoInt <- FindVariableFeatures(iPSCsDopa_WoInt, selection.method = "vst", nfeatures = 3000)


all.genes <- rownames(iPSCsDopa_WoInt)
iPSCsDopa_WoInt <- ScaleData(iPSCsDopa_WoInt, features = all.genes)



iPSCsDopa_WoInt <- RunPCA(iPSCsDopa_WoInt, features = VariableFeatures(object = iPSCsDopa_WoInt), npcs=30)






iPSCsDopa_WoInt <- FindNeighbors(iPSCsDopa_WoInt,reduction = "pca", dims = 1:30, nn.eps = 0.5)
iPSCsDopa_WoInt <- FindClusters(iPSCsDopa_WoInt, resolution = 0.1, n.start = 10)

iPSCsDopa_WoInt <- RunUMAP(iPSCsDopa_WoInt, dims = 1:30)

p2 <- DimPlot(iPSCsDopa_WoInt, reduction = "umap", label = TRUE)

pdf(paste(pathto.outPlots,"UMAP_Wointegrated.pdf"), width=10, height=10)
plot_grid(p2)
dev.off()


NumberOfPerSampleCellsInCluster <- as.list(table(paste0(iPSCsDopa_WoInt$SampleID,";",iPSCsDopa_WoInt$seurat_clusters)))

NumberOfPerSampleCellsInClusterDf <- (data.frame(matrix(nrow=length(unique(iPSCsDopa_WoInt$SampleID)),ncol=length(unique(iPSCsDopa_WoInt$seurat_clusters)))))

rownames(NumberOfPerSampleCellsInClusterDf) <- unique(iPSCsDopa_WoInt$SampleID)

colnames(NumberOfPerSampleCellsInClusterDf) <- paste0("Cluster",sort(unique(iPSCsDopa_WoInt$seurat_clusters)))


for (RowNum in 1:nrow(NumberOfPerSampleCellsInClusterDf)){
  
  
  for (ColNum in 1:ncol(NumberOfPerSampleCellsInClusterDf)){
    ValToPut <- NumberOfPerSampleCellsInCluster[match(gsub("Cluster","",paste0(rownames(NumberOfPerSampleCellsInClusterDf)[RowNum],";",
                                                                               colnames(NumberOfPerSampleCellsInClusterDf)[ColNum])),names(NumberOfPerSampleCellsInCluster))]
    if(is.numeric(unlist(ValToPut))){
      NumberOfPerSampleCellsInClusterDf[RowNum,ColNum] <-ValToPut
    } else {NumberOfPerSampleCellsInClusterDf[RowNum,ColNum] <- 0}
    
  }
  
}

NumberOfPerSampleCellsInClusterDfV2 <- NumberOfPerSampleCellsInClusterDf

NumberOfPerSampleCellsInClusterDf$TotalCells <- table(iPSCsDopa_WoInt$SampleID)




NumberOfPerSampleCellsInClusterDfV2Percent <- (NumberOfPerSampleCellsInClusterDfV2/NumberOfPerSampleCellsInClusterDf$TotalCells)*100




NumberOfPerSampleCellsInClusterDfV2PercentV2 <- NumberOfPerSampleCellsInClusterDfV2Percent
colnames(NumberOfPerSampleCellsInClusterDfV2PercentV2) <- paste0(colnames(NumberOfPerSampleCellsInClusterDfV2Percent)," (",table(iPSCsDopa_WoInt$seurat_clusters),")")


pdf(paste(pathto.outPlots,"Heatmap_PercentOfPerSampleCellsInCluster_Wointegrated.pdf"), width=10, height=10)

#par(mar=c(17,14,14,12)) 
heatmap(as.matrix(t(NumberOfPerSampleCellsInClusterDfV2PercentV2)), Rowv=NA, Colv=NA, margins = c(15,10))

dev.off()

rm(list = c('iPSCsDopa_WoInt'))


NumberOfPerSampleCellsInClusterDfV3 <- t(NumberOfPerSampleCellsInClusterDfV2)
NumberOfPerClusterCellsFromSamplesDfV2Percent <- (t((NumberOfPerSampleCellsInClusterDfV3/unlist(colSums(NumberOfPerSampleCellsInClusterDfV2)))*100))

pdf(paste(pathto.outPlots,"Heatmap_PercentOfPerClusterCellsFromSamples_Wointegrated.pdf"), width=10, height=10)

#par(mar=c(17,14,14,12)) 
heatmap(as.matrix(t(NumberOfPerClusterCellsFromSamplesDfV2Percent)), Rowv=NA, Colv=NA, margins = c(15,10))

dev.off()



writeLines(capture.output(sessionInfo()), paste(pathto.outData,"sessionInfo.txt"))

#rm(list=setdiff(ls(), c("iPSCsDopa.integrated","iPSCsDopa.markers")))




